File: OOP/1D & 2D Arrays/BubbleSort.java
import java.util.*;

class BubbleSort{

	private static int i,j;
	private static Scanner sc = new Scanner(System.in);

	public static void main(String[] args){
		double[] a;
		int s;
		a = ReadArray();
		a = ascending(a);
		System.out.println("Sort in\n1. Ascending Order\n2. Descending Order\n");
		s = sc.nextInt();
		switch(s){
			case 1:
			a = ascending(a);
			case 2:
			a = descending(a);
			default:
			System.out.println("Enter valid option");
		}
		PrintArray(a);
	}

	private static double[] ReadArray(){
		System.out.println("Enter number of elements in array");
		int n = sc.nextInt();
		double[] arr = new double[n];
		System.out.println("Enter elements in array");
		for(i=0; i<arr.length; i++){
			arr[i] = sc.nextDouble();
		}
		return arr;
	}

	private static void PrintArray(double[] arr){
		for(double i:arr){
			System.out.println(i);
		}
	}

	private static double[] ascending(double[] arr){
		int i,j;
		double temp;
		for(i=0; i<arr.length-1; i++){
			for(j=i+1;j<=arr.length-1; j++){
				if(arr[j]<arr[i]){
					temp = arr[j];
					arr[j] = arr[i];
					arr[i] = temp;
				}
			}
		}
		return arr;
	}

	private static double[] descending(double[] arr){
		int i,j;
		double temp;
		for(i=0; i<arr.length-1; i++){
			for(j=i; j<=arr.length-1; j++){
				if(arr[j]>arr[i]){
					temp = arr[j];
					arr[j] = arr[i];
					arr[i] = temp;
				}
			}
		}
		return arr;
	}
}

File: ALG/brute_force_I/lab_exercise/bubble_sort.c
#include <stdio.h>

void swap(int* a, int* b) {
	int temp = *a;
	*a = *b;
	*b = temp;
}

int* bubble_sort(int arr[], int n) {
	for (int k=0; k<n-1; k++) {
		for (int i=k; i<n-1; i++) {
			if (arr[i] > arr[i+1])
				swap(&arr[i], &arr[i+1]);
		}
	}
	return arr;
}

int main(void) {
	int n;
	printf("Enter length of array:\n");
	scanf("%d", &n);
	int arr[n];
	printf("Enter numbers\n");
	for (int i=0; i<n; i++)
		scanf("%d", &arr[i]);
	int* sorted;
	sorted = bubble_sort(arr, n);
	printf("Sorted:\n");
	for (int j=0 ; j<n; j++)
		printf("%d\n", sorted[j]);
}